backend/de.metas.business/src/main/java/de/metas/product/impl/ProductDAO.java

```diff
@@ -769,4 +772,29 @@ else if (valueType == IssuingToleranceValueType.QUANTITY)
 			throw new AdempiereException("Unknown valueType: " + valueType);
 		}
 	}
+
+	@Override
+	public boolean isProductUsed(@NonNull final ProductId productId)
```

Comment:
IMHO this shall go to ProductBL.
It's not ProductDAO because there is no relation with Product tables.

OK, ideally would be to have "handlers" injected, each of them coming from different "modules" like invoice, shipments etc

Reply:
also is it OK that we consider even draft lines?
(I guess yes, but just want to be sure)

Reply:
also, what about inventory, movements, cost collectors, cost details?

Reply:
Also it's checked if an uomConversion exists before, so maybe we don't have Problems in other areas then